lib/molinillo/resolution.rb

```diff
@@ -326,16 +340,15 @@ def unwind_details_for_requirements(binding_requirements, existing_unwind_detail
           until grandparent_r.nil?
             requirement_state = find_state_for(grandparent_r)
             candidate_index = states.index(requirement_state)
-            break unless candidate_index && candidate_index >= unwind_details.state_index
-            if requirement_state && !requirement_state.possibilities.empty?
-              unwind_details = UnwindDetails.new(candidate_index, r, :grandparent, binding_requirements)
-              break
+            unless requirement_state.possibilities.empty?
+              unwind_details <<
+                UnwindDetails.new(candidate_index, requirement_tree_for(r), :grandparent, binding_requirements)
             end
             grandparent_r = parent_of(grandparent_r)
           end
         end
 
-        unwind_details
+        unwind_details.sort.reverse
```

Comment:
might it be faster to do `unwind_details.sort { |l ,r| r <=> l }` to avoid the intermediary array?